let User,CustomErrorHandler;_e14‍.x([["default",()=>_e14‍.o]]);_e14‍.w("../../models",[["User",["User"],function(v){User=v}]]);_e14‍.w("../../services",[["CustomErrorHandler",["CustomErrorHandler"],function(v){CustomErrorHandler=v}]]);


const userController = {
    async getUserDetails(req, res, next) {
        try {
            const user = await User.findOne({ _id: req.user._id }).select('-password -updatedAt -__v');
            _e14‍.g.console.log("User data:", user);
            if (!user) {
                return next(CustomErrorHandler.notFound());
            }
            res.json({
                id: user._id,
                name: user.name,
                email: user.email,
                createdAt: user.createdAt
            });
        } catch (err) {
            return next(err);
        }
    }
};

_e14‍.d(userController);